--- /core/iwasm/interpreter/wasm_interp_fast.c.orig	
+++ /core/iwasm/interpreter/wasm_interp_fast.c
@@ -3981,10 +3981,10 @@
 
     RECORD_STACK_USAGE(exec_env, (uint8 *)&prev_frame);
-#if !(defined(OS_ENABLE_HW_BOUND_CHECK) \
+#if (defined(OS_ENABLE_HW_BOUND_CHECK) \
       && WASM_DISABLE_STACK_HW_BOUND_CHECK == 0)
     if ((uint8 *)&prev_frame < exec_env->native_stack_boundary) {
         wasm_set_exception((WASMModuleInstance *)exec_env->module_inst,
                            "native stack overflow");
         return;
     }
 #endif

--- /core/shared/platform/common/posix/posix_memmap.c
+++ /core/shared/platform/common/posix/posix_memmap.c
@@ -76,12 +76,12 @@ os_mmap(void *hint, size_t size, int prot, int flags, os_file_handle file)
     if (prot & MMAP_PROT_EXEC)
         map_prot |= PROT_EXEC;
 
-#if defined(BUILD_TARGET_X86_64) || defined(BUILD_TARGET_AMD_64)
-#ifndef __APPLE__
-    if (flags & MMAP_MAP_32BIT)
-        map_flags |= MAP_32BIT;
-#endif
-#endif
+    // #if defined(BUILD_TARGET_X86_64) || defined(BUILD_TARGET_AMD_64)
+    // #ifndef __APPLE__
+    //     if (flags & MMAP_MAP_32BIT)
+    //         map_flags |= MAP_32BIT;
+    // #endif
+    // #endif
 
     if (flags & MMAP_MAP_FIXED)
         map_flags |= MAP_FIXED;